src/Server/LanguageServer.cs

```diff
@@ -301,38 +302,26 @@ public IDisposable AddTextDocumentIdentifier(params ITextDocumentIdentifier[] ha
             return _textDocumentIdentifiers.Add(ActivatorUtilities.CreateInstance<T>(_serviceProvider));
         }
 
-        private IDisposable RegisterHandlers(LspHandlerDescriptorDisposable handlerDisposable)
+        public async Task Initialize(CancellationToken token)
         {
-            using (var scope = _serviceProvider.CreateScope())
+            if (_initializingTask != null)
             {
-                var registrations = handlerDisposable.Descriptors
-                    .Where(d => d.AllowsDynamicRegistration)
-                    .Select(d => new Registration() {
-                        Id = d.Id.ToString(),
-                        Method = d.Method,
-                        RegisterOptions = d.RegistrationOptions
-                    })
-                    .ToArray();
-
-                // Fire and forget
-                DynamicallyRegisterHandlers(registrations).ToObservable().Subscribe();
+                try
+                {
+                    await _initializingTask;
+                }
+                catch
+                {
+                    // Swallow exceptions because the original initialization task will report errors if it fails (don't want to doubly report).
+                }
 
-                return new ImmutableDisposable(
-                    handlerDisposable,
-                    Disposable.Create(() => {
-                        Client.UnregisterCapability(new UnregistrationParams() {
-                            Unregisterations = registrations.ToArray()
-                        }).ToObservable().Subscribe();
-                    }));
+                return;
             }
-        }
 
-        private async Task Initialize(CancellationToken token)
-        {
-            _connection.Open();
             try
             {
-                await _initializeComplete
+                _connection.Open();
```

Comment:
If the connection is already opened does this have any effect?

Reply:
If the connection is already open and you attempt to call open again you'll get a:

![image](https://user-images.githubusercontent.com/2008729/75376355-c1e29200-5884-11ea-8181-59e22f2e2fa4.png)

Because this is effectively what calling open does.

That being said, how would one get access to the `_connection` object to call open twice? I ask because at the top of this method I ensure this `_connection.Open` can't be called more than once.

Reply:
I'm slow I didn't see the old connection open call in the original diff.